lib/Service/OpenProjectAPIService.php

```diff
@@ -886,16 +886,45 @@ public function deleteFileLink(int $fileLinkId, string $userId): array {
 		return $result;
 	}
 
+	/**
+	 * check common admin settings
+	 * @param IConfig $config
+	 *
+	 * @return bool
+	 */
+	public static function isCommonAdminConfigOk(IConfig $config): bool {
+		$oauthInstanceUrl = $config->getAppValue(Application::APP_ID, 'openproject_instance_url');
+		$freshProjectFolderSetUp = (bool)$config->getAppValue(Application::APP_ID, 'fresh_project_folder_setup');
+
+		if ($freshProjectFolderSetUp === true || empty($oauthInstanceUrl) || !self::validateURL($oauthInstanceUrl)) {
+			return false;
+		}
+
+		return true;
+	}
+
 	/**
 	 * checks if every admin config for oauth2 based authorization variables are set
 	 *
 	 * @param IConfig $config
 	 * @return bool
 	 */
 	public static function isAdminConfigOkForOauth2(IConfig $config):bool {
-		$clientId = $config->getAppValue(Application::APP_ID, 'openproject_client_id');
-		$clientSecret = $config->getAppValue(Application::APP_ID, 'openproject_client_secret');
-		return !(empty($clientId) || empty($clientSecret));
+		if (!self::isCommonAdminConfigOk($config)) {
+			return false;
+		}
+
+		$authMethod = $config->getAppValue(Application::APP_ID, 'authorization_method');
+		// NOTE: For backwards compability, check the auth method only if provided
+		// version: 2.8 -> 2.9
+		if ($authMethod && $authMethod !== self::AUTH_METHOD_OAUTH) {
+			return false;
+		}
+
+		$opClientId = $config->getAppValue(Application::APP_ID, 'openproject_client_id');
+		$opClientSecret = $config->getAppValue(Application::APP_ID, 'openproject_client_secret');
+		$ncClientId = $config->getAppValue(Application::APP_ID, 'nc_oauth_client_id');
```

Comment:
```suggestion
		$ncClientId = $config->getAppValue(Application::APP_ID, 'nc_oauth_client_id');
```

why `ncClientId` need to be checked? 
If this need to be checked why not `nc Oauth Secret` ?